### PR TITLE
refactors frame create test

### PR DESCRIPTION
**Pull request by Haleyo**

### CODE REVIEW

File: regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    def test_frame_invalid_column(self):
+        """Tests retrieving an invalid column errors."""
+        with self.assertRaises(ValueError):
+            self.frame.take(100, columns=['not_in'])
+
+    def test_frame_create_row_count(self):
+        """ Trivial Frame creation. """
+        frame = self.context.frame.import_csv(self.dataset, schema=self.schema)
+        self.assertEqual(frame.row_count, 3)
+        self.assertEqual(len(frame.take(3).data), 3)
```
Comment (blbarker) [inline regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py:29]: The take should try to take much more, else the test misses that take only finds 3 rows.
And, especially with only 3 rows, we should just compare the data and see that it is appropriate

File: regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        dataset_passwd = self.get_file("passwd.csv")
+        passwd_schema = [("userid", str),
+                         ("password", str),
+                         ("usrNum", int),
+                         ("grpNum", int),
+                         ("userName", str),
+                         ("home", str),
+                         ("shell", str)]
+        passwd_frame = self.context.frame.import_csv(dataset_passwd, schema=passwd_schema, delimiter=':')
+        self.assertEqual(len(passwd_frame.take(1).data[0]), len(passwd_schema))
```
Comment (blbarker) [inline regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py]: Need to compare data while we're at it.  Comparing len is not getting much coverage.

File: regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    def test_frame_delim_tab(self):
+        """Test building a frame with a tab delimiter."""
+        dataset_delimT = self.get_file("delimTest1.tsv")
+        white_schema = [("col_A", int),
+                        ("col_B", long),
+                        ("col_3", float),
+                        ("Double", float),
+                        ("Text", str)]
+        frame = self.context.frame.import_csv(dataset_delimT, schema=white_schema, delimiter='\t')
+        self.assertEqual(len(frame.take(1).data[0]), len(white_schema))
```
Comment (blbarker) [inline regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py]: Again, compare data

File: regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        """Test an empty delimiter errors."""
+        with self.assertRaisesRegexp(Exception, "delimiter"):
+            self.context.frame.import_csv(self.dataset, self.schema, delimiter="")
+
+    def test_header(self):
+        """Test the baseline number of row without modification is correct."""
+        frame_with_header = self.context.frame.import_csv(
+            self.dataset, schema=self.schema, header=True)
+        frame_without_header = self.context.frame.import_csv(self.dataset, schema=self.schema, header=False)
+        self.assertEqual(len(frame_with_header.take(sys.maxint).data), len(frame_without_header.take(sys.maxint).data) - 1)
```
Comment (blbarker) [inline regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py]: Again, compare the data.  What if it dropped the last row and not the header?  I know, an unlikely what-if, but we should tighten the coverage when going to the effort of setting up the testcase.

**Comment by Haleyo** [comment]
Updated frame create test from your suggestions Briton

**Pull request merged**